hal/src/rtl872x/interrupts_hal.cpp

```diff
@@ -150,9 +207,9 @@ int hal_interrupt_attach(uint16_t pin, hal_interrupt_handler_t handler, void* da
             return SYSTEM_ERROR_INVALID_ARGUMENT;
         }
 
+        GPIO_Init_HAL(&GPIO_InitStruct, pinInfo->gpio_port);
         GPIO_UserRegIrq(rtlPin, (VOID*)gpioIntHandler, (void*)((uint32_t)pin));
-
-        GPIO_INTMode(rtlPin, ENABLE, GPIO_InitStruct.GPIO_ITTrigger, GPIO_InitStruct.GPIO_ITPolarity, GPIO_INT_DEBOUNCE_ENABLE);
+        GPIO_INTMode_HAL(rtlPin, pinInfo->gpio_port, ENABLE, GPIO_InitStruct.GPIO_ITTrigger, GPIO_InitStruct.GPIO_ITPolarity, GPIO_INT_DEBOUNCE_ENABLE);
```

Comment:
Why do we call `GPIO_INTMode_HAL` twice here? Once in `GPIO_Init_HAL` once here.

Reply:
It's called this way because of the Realtek SDK docs, and all SDK examples.  [See here](https://github.com/particle-iot/realtek_ambd_sdk_public/blob/main/component/soc/realtek/amebad/fwlib/include/rtl8721d_gpio.h#L59-L82)  Maybe this isn't fully correct, but I didn't want to make any assumptions since I don't know everything that `GPIO_UserRegIrq()` does.